src/project/project.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Project } from './project.entity';
+import { CreateProjectDto } from './dto/create-project.dto';
+import { UpdateProjectDto } from './dto/update-project.dto';
+
+@Injectable()
+export class ProjectService {
+  constructor(
+    @InjectRepository(Project)
+    private projectRepository: Repository<Project>,
+  ) {}
+
+  async createProject(createProjectDto: CreateProjectDto): Promise<Project> {
+    const project = this.projectRepository.create(createProjectDto);
+    return this.projectRepository.save(project);
+  }
+
+  async getProjectById(id: number): Promise<Project> {
+    const project = await this.projectRepository.findOne({ where: { id } });
+    if (!project) {
+      throw new NotFoundException(`Project with ID ${id} not found`);
+    }
+    return project;
+  }
+
+  async getAllProjects(): Promise<Project[]> {
+    return this.projectRepository.find();
+  }
+
+  async updateProject(
```

Comment:
(praise): I love to see a REST API that returns the updated object. It's super helpful for front-end clients by reducing the number of requests made to the API. Nicely done.